devbin/generate_gcp_ar_cleanup_policy.py

```diff
@@ -0,0 +1,140 @@
+import abc
+import json
+import yaml
+
+from typing import List, Optional
+
+
+class CleanupPolicy(abc.ABC):
+    @abc.abstractmethod
+    def to_dict(self):
+        pass
+
+
+class DeletePolicy(CleanupPolicy):
+    def __init__(self,
+                 name: str,
+                 tag_state: str,
+                 *,
+                 tag_prefixes: Optional[List[str]] = None,
+                 version_name_prefixes: Optional[List[str]] = None,
+                 package_name_prefixes: Optional[List[str]] = None,
+                 older_than: Optional[str] = None,
+                 newer_than: Optional[str] = None):
+        self.name = name
+        self.tag_state = tag_state
+        self.tag_prefixes = tag_prefixes
+        self.version_name_prefixes = version_name_prefixes
+        self.package_name_prefixes = package_name_prefixes
+        self.older_than = older_than
+        self.newer_than = newer_than
+
+    def to_dict(self):
+        data = {'name': self.name, 'action': {'type': 'Delete'}, 'condition': {'tagState': self.tag_state}}
+        condition = data['condition']
+        if self.tag_prefixes is not None:
+            condition['tagPrefixes'] = self.tag_prefixes
+        if self.version_name_prefixes is not None:
+            condition['versionNamePrefixes'] = self.version_name_prefixes
+        if self.package_name_prefixes is not None:
+            condition['packageNamePrefixes'] = self.package_name_prefixes
+        if self.older_than:
+            condition['olderThan'] = self.older_than
+        if self.newer_than:
+            condition['newerThan'] = self.newer_than
+        return data
+
+
+class ConditionalKeepPolicy(CleanupPolicy):
```

Comment:
This is just a `KeepPolicy`, right? Or, otherwise, the above is a `ConditionalDeletePolicy`?

Reply:
I just used their nomenclature in the docs